### PR TITLE
Support on-premise s3-compatible storage.

### PR DESCRIPTION
**Pull request by acmore**
Support on-premise s3-compatible storage. User can now pass an instance of pyarrow filesystem to `make_reader` and `make_batch_reader` functions.

### CODE REVIEW

File: petastorm/reader.py [hunk 1]
```diff
@@ -123,14 +124,16 @@ def make_reader(dataset_url,
         here: https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetDataset.html
     :param s3_config_kwargs: dict of parameters passed to ``botocore.client.Config``
     :param zmq_copy_buffers: A bool indicating whether to use 0mq copy buffers with ProcessPool.
+    :param filesystem: A filesystem to use. Will ignore s3_config_kwargs and other filesystem configs if it's provided.
```
Comment (selitvin) [inline petastorm/reader.py:127]: Let's mention that this is a pyarrow filesystem is expected.

File: docs/release-notes.rst [hunk 2]
```diff
@@ -8,6 +8,7 @@ Release notes
 Release 0.9.8 (unreleased)
 ==========================
 - `PR 617 <https://github.com/uber/petastorm/pull/617>`_: Deprecating `pyarrow_serialize` argument of `petastorm.make_reader`.
+- `PR 625 <https://github.com/uber/petastorm/pull/625>`_: Support on-premise s3-compatible storage.
```
Comment (selitvin) [inline docs/release-notes.rst:11]:
```suggestion
- `PR 625 <https://github.com/uber/petastorm/pull/625>`_: Support on-premise s3-compatible storage.
- `PR 625 <https://github.com/uber/petastorm/pull/625>`_: Support on-premise s3-compatible storage. User can now pass an instance of pyarrow filesystem to `make_reader` and `make_batch_reader` functions.
```

**Review by selitvin — COMMENTED** [review]
Yep. This is handy. Thank you for the PR! 
Minor comments. Can you please update the PR summary with more information please?

File: petastorm/reader.py (hunk 1 above)
Comment (acmore) [reply petastorm/reader.py]: Sure.

**Comment by selitvin** [comment]
For the PR to land, please:
- rebase
- fix linter issue/s
- make sure tests pass

**Comment by acmore** [comment]
> For the PR to land, please:
> 
> * rebase
> * fix linter issue/s
> * make sure tests pass

Sure. All the tests passed.

File: docs/release-notes.rst [hunk 3]
```diff
@@ -8,6 +8,8 @@ Release notes
...
+- `PR 625 <https://github.com/uber/petastorm/pull/625>`_: Support on-premise s3-compatible storage. User can now pass an instance of pyarrow filesystem to `make_reader` and `make_batch_reader` functions.
```
Comment (selitvin) [inline docs/release-notes.rst:12]: Duplicate line.

**Review by selitvin — APPROVED** [review]

**Comment by selitvin** [comment]
Can you please populate the PR comment?

**Comment by acmore** [comment]
> Can you please populate the PR comment?

Oh sorry. I didn't notice about it. Resolved.

**Comment by selitvin** [comment]
Excellent. Thank you for your PR. I'll cut a release within couple of days.

**Pull request merged**